tests/test_script_production.py

```diff
@@ -1356,3 +1361,256 @@ def normpath(path):
                     ).replace("/", ":NORM:"),
                 ],
             )
+
+
+class RecursiveMixin:
+    rev: List[str]
+    rev_to_script: Dict[str, Script]
+    org_script_dir: ScriptDirectory
+
+    def _move_revision_file(
+        self,
+        rev_index: int,
+        destination_dir: str,
+        version_path="scripts/versions",
+    ):
+        rev_hash = self.rev[rev_index]
+        script = self.rev_to_script[rev_hash]
+        target_file = self._get_moved_path(
+            rev_index, destination_dir, version_path
+        )
+        os.makedirs(os.path.dirname(target_file), exist_ok=True)
+        shutil.move(script.path, target_file)
+
+    def _get_moved_path(
+        self,
+        rev_index: int,
+        destination_dir: str = "",
+        version_path="scripts/versions",
+    ):
+        rev_hash = self.rev[rev_index]
+        script = self.rev_to_script[rev_hash]
+        file_name = os.path.basename(script.path)
+        target_file = os.path.join(
+            _get_staging_directory(), version_path, destination_dir, file_name
+        )
+        target_file = os.path.realpath(target_file)
+        return target_file
+
+    def _checkTopologicalOrder(self, sd: ScriptDirectory):
```

Comment:
**mike bayer** (zzzeek) wrote:

nit: replace _lowerCamelCase with _underscore_and_dashes ; I guess the convention here is from python unittest but we aren't standardizing on that

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/4457

Reply:
I use daily camelCase, so I must overlook this  - fixed.